### DISCUSSION TITLE
Dates comparison by isDataEqual

### DISCUSSION THREAD

**Discussion by cherniavskii** [TanStack/query#433] [General] [answered]
Hi :)

In 1.3.0 `deepEqual` is used for data comparison.

I've noticed that `Date` objects are not properly compared, since comparing two different dates always returns `true`.
While this is edge case and APIs always return serialized data, I wonder if this should be fixed.

Theoretically, `queryFn` can return virtually anything and doesn't have to use any API at all.

What do you think?

**Comment by tannerlinsley** [comment, marked as answer]
I think dates are a safe one to include, mostly because json.parse will sometimes auto create dates from strings

**Reply by cherniavskii** [reply]
What do you think about adding `fast-deep-equal` as dependency?
There's 4 additional lines of code in original `fast-deep-equal`, and dates support is one of them.
Also, we wouldn't need to write tests for it

**Comment by tannerlinsley** [comment]
How about we inline it instead. Saves .2kb ;)
